lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -379,6 +379,15 @@ void FrontendInputsAndOutputs::forEachOutputFilename(
       });
 }
 
+void FrontendInputsAndOutputs::forEachOutputFilename(
+    llvm::function_ref<void(StringRef, file_types::ID)> fn) const {
+  (void)forEachInputProducingAMainOutputFile(
+      [&](const InputFile &input) -> bool {
+        fn(input.outputFilename(), input.getType());
```

Comment:
Could be wrong, but I think `getType()` here gets the type of the input file, not the type of the output file.